question-servers/freeformPythonLib/prairielearn.py

```diff
@@ -578,22 +578,32 @@ def string_partition_outer_interval(s, left='[', right=']'):
     return s_before_left, s, s_after_right
 
 
-def string_to_integer(s):
-    """string_to_integer(s)
+def string_to_integer(s, base=10):
+    """string_to_integer(s, base=10)
 
     Parses a string that is an integer.
 
     Returns a number with type int, or None on parse error.
     """
+    if s is None:
+        return None
+
     # Replace unicode minus with hyphen minus wherever it occurs
     s = s.replace(u'\u2212', '-').strip()
+
+    # If base is 16, strip "0x" prefix
+    if base == 16 and s.startswith('0x'):
+        s = s[2:].strip()
+    elif base == 16 and s.startswith('-0x'):
+        s = '-' + s[3:].strip()
+
```

Comment:
Can you explain why this section is necessary? AFAICT, `int()` handles this for us:

```python
>>> int('FFFFFF', 16)
16777215
>>> int('0xFFFFFF', 16)
16777215
>>> int('-FFFFFF', 16)
-16777215
>>> int('-0xFFFFFF', 16)
-16777215
```

Reply:
Good point, I didn't realize that. In light of that I removed this piece of code, but I also added an option for special base value 0.

Reply:
What does it mean for a number to be base 0? I've never encountered that idea before - how would that work?

Reply:
Ah, I should read the docs you wrote, shouldn't I...

Would it make sense for that to be the default behavior no matter what base is specified? i.e., even if `base="7"`, I could still write "0x123abc" as an answer?

Reply:
I would prefer to use base=0 for that, otherwise you may have problems with inputs like 0777. Most students would expect this to mean base 10, except when you explicitly allow for it to be base 8.

Reply:
and if you need the base to be something other than 10, you probably want to force students to use that base as part of the question...

Reply:
Makes sense to me.